scripts/verify-packaged-fork-config.sh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+if [ $# -lt 2 ]; then
+    echo "Usage: $0 <mainnet-fork-config.json> <working-dir>"
+    cat <<EOF
+This script is used to validate that an installed package is correct
+according to an exported fork_config.json file.
+
+Inputs:
+- The exported mainnet full config.json fork config, with all accounts
+- A working directory where ledgers/configs will be created
+- Installed MINA_EXE (default: mina) and MINA_GENESIS_EXE (default: mina-create-genesis) programs
+- PACKAGED_CONFIG_JSON (default: /var/lib/coda/config_*.json)
+  the runtime config generated by the HF packaging
+- GENESIS_LEDGER_DIR (default: /var/lib/coda)
+  where the tarfiles are stored
+- FORKING_FROM_CONFIG_JSON (default: /var/lib/coda/mainnet.json)
+  the pre-fork genesis ledger
+- SECONDS_PER_SLOT (default: 180)
+
+Ensures:
+- The accounts listed in config.json are the ones in the PACKAGED_CONFIG_JSON
+- The genesis ledger directory tarfile contents match reference copies created
+  fresh from config.json. This takes a long time (>20min) due to rehashing.
+
+Outputs:
+- Exit code 0 if validated, 1 otherwise.
+EOF
+    
+    exit 1
+fi
+
+MINA_EXE=${MINA_EXE:-"mina"}
+MINA_GENESIS_EXE=${MINA_GENESIS_EXE:-"mina-create-genesis"}
+found_config=$(echo /var/lib/coda/config_*.json)
+PACKAGED_CONFIG_JSON=${PACKAGED_CONFIG_JSON:-$found_config}
+GENESIS_LEDGER_DIR=${GENESIS_LEDGER_DIR:-"/var/lib/coda"}
+SECONDS_PER_SLOT=${SECONDS_PER_SLOT:=180}
+FORKING_FROM_CONFIG_JSON=${FORKING_FROM_CONFIG_JSON:=/var/lib/coda/mainnet.json}
+
+export MINA_LIBP2P_PASS=''
+
+workdir=$2
+mkdir -p "$workdir"
+mkdir -p "$workdir/ledgers"
+mkdir -p "$workdir/ledgers-backup"
+mkdir -p "$workdir/keys"
+chmod 700 "$workdir/keys"
+
+if [ ! -e "$workdir/keys/p2p" ]; then
+    "$MINA_EXE" libp2p generate-keypair --privkey-path "$workdir/keys/p2p"
+fi
+
+# inspired by scripts/hardfork/create_runtime_config.sh
+function generate_ledgers_tar_from_config() {
```

Comment:
Why is the separate from the existing script? Is there a way we can better share code here?

Reply:
I tried to contain this script to a single file to make it easy to use with or without a source directory, bundled as part of a package or run in-tree.